### PR TITLE
Update link in options.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Update GNU ARM Embedded link to point to latest update

### CODE REVIEW

File: docs/dev_tools/options.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ We created the mbed command-line tool (mbed CLI), a Python-based tool, specifica
 mbed OS 5 can be built with various toolchains. The currently supported versions are:
 
 * [ARM compiler 5.06 update 3](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler-5/downloads).
-* [GNU ARM Embedded version 4.9](https://launchpad.net/gcc-arm-embedded/4.9/4.9-2015-q3-update).
+* [GNU ARM Embedded version 6-2017-q1-update](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads).
```
Comment (sg-) [inline docs/dev_tools/options.md]: Just GNU ARM Embedded version 6 should be fine.

File: docs/dev_tools/options.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ We created the mbed command-line tool (mbed CLI), a Python-based tool, specifica
 mbed OS 5 can be built with various toolchains. The currently supported versions are:
 
 * [ARM compiler 5.06 update 3](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler-5/downloads).
-* [GNU ARM Embedded version 4.9](https://launchpad.net/gcc-arm-embedded/4.9/4.9-2015-q3-update).
+* [GNU ARM Embedded version 6](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads).
 * [IAR Embedded Workbench 7.7](https://www.iar.com/iar-embedded-workbench/tools-for-arm/arm-cortex-m-edition/).
```
Comment (ashok-rao) [inline docs/dev_tools/options.md]: Shouldn't IAR be 7.8 ? Or is it not official yet?

File: docs/dev_tools/options.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ We created the mbed command-line tool (mbed CLI), a Python-based tool, specifica
 mbed OS 5 can be built with various toolchains. The currently supported versions are:
 
 * [ARM compiler 5.06 update 3](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler-5/downloads).
```
Comment (ashok-rao) [inline docs/dev_tools/options.md]: 5.06 Update 4 ?

File: docs/dev_tools/options.md (hunk 2 above)
Comment (sg-) [inline docs/dev_tools/options.md]: ARM 5 is 5.06 update 3 or greater
IAR is 7.7 or greater currently
GCC is 5.4 or greater and 6 recommended

Comment (ashok-rao) [inline docs/dev_tools/options.md]: Super. Thanks Sam.
@AnotherButler , please ignore my earlier comments. Happy for you to merge this.

Comment (AnotherButler) [inline docs/dev_tools/options.md]: Awesome. Thanks so much.

**Pull request merged**